packages/ckeditor5-table/src/tablelayout/tablelayoutui.ts

```diff
@@ -116,5 +134,94 @@ export default class TableLayoutUI extends Plugin {
 
 			return menuView;
 		} );
+
+		this._registerTableTypeSwitchButton();
+	}
+
+	/**
+	 * Registers the table type switch button. It overrides the default table properties button
+	 * with a dropdown that allows the user to choose between different table types.
+	 *
+	 * @internal
+	 */
+	private _registerTableTypeSwitchButton(): void {
+		const editor = this.editor;
+		const t = editor.t;
+		const { componentFactory } = editor.ui;
+		const tablePropertiesUI = editor.plugins.get( TablePropertiesUI );
+
+		// Override the tableProperties button with a dropdown.
+		componentFactory.add( 'tableProperties', locale => {
+			const originalButton = tablePropertiesUI._createTablePropertiesButton();
+
+			// Wrap the original button in a SplitButtonView.
+			const dropdownButton = new SplitButtonView( locale, originalButton );
+			const dropdownView = createDropdown( locale, dropdownButton );
+			const itemsDefinitions = createTableLayoutTypeDropdownItems( editor );
+
+			// Add table types to the dropdown.
+			addListToDropdown( dropdownView, itemsDefinitions, {
+				ariaLabel: t( 'Table type options' )
+			} );
+
+			dropdownButton.tooltip = t( 'Choose table type' );
+			dropdownView.on<ButtonExecuteEvent>( 'execute', evt => {
+				if ( typeof ( evt.source as any )._action === 'function' ) {
+					( evt.source as any )._action();
+				}
+			} );
+
+			return dropdownView;
+		} );
 	}
 }
+
+/**
+ * Creates dropdown items for table type selection.
+ *
+ * @param editor The editor instance.
+ * @returns A collection of dropdown items for the table type dropdown.
+ * @internal
+ */
+function createTableLayoutTypeDropdownItems( editor: Editor ) {
+	const t = editor.t;
+	const tableTypeCommand = editor.commands.get( 'tableType' )!;
+	const itemDefinitions = new Collection<ListDropdownButtonDefinition>();
+
+	itemDefinitions.add( createTableTypeDropdownItem( tableTypeCommand, 'layout', t( 'Layout table' ) ) );
+	itemDefinitions.add( createTableTypeDropdownItem( tableTypeCommand, 'content', t( 'Content table' ) ) );
```

Comment:
Labels missing in context.json. Should it be "Table layout" as such context is already defined? Maybe we should align all references to the same.